File: server/shared/shapes.py
import math
from enum import Enum



class Shape(Enum):
    # CIRCLE = "circle"
    SQUARE = "square"
    TRIANGLE = "triangle"
    HEXAGON = "hexagon"


def regular_polygon(sides, radius):
    return [
        (
            radius * math.cos(2 * math.pi * i / sides),
            radius * math.sin(2 * math.pi * i / sides)
        )
        for i in range(sides)
    ]

SHAPES = {
    "triangle": [(-10, 10), (10, 10), (0, -10)],
    "square": [(-10, -10), (10, -10), (10, 10), (-10, 10)],
    "hexagon": regular_polygon(6, 10),
}

File: client/shared/shapes.js
export const SHAPES = {
  triangle: [
    [-10, 10],
    [10, 10],
    [0, -10],
  ],
  square: [
    [-10, -10],
    [10, -10],
    [10, 10],
    [-10, 10],
  ],
  hexagon: Array.from({ length: 6 }, (_, i) => {
    const angle = (Math.PI * 2 * i) / 6;
    return [Math.cos(angle) * 10, Math.sin(angle) * 10];
  }),
};
